components/brave_wallet/browser/cardano/cardano_hd_keyring.cc

```diff
@@ -115,6 +86,27 @@ std::optional<std::string> CardanoHDKeyring::AddNewHDAccount(uint32_t index) {
   return "";
 }
 
+std::optional<std::array<uint8_t, kCardanoSignatureSize>>
+CardanoHDKeyring::SignMessage(uint32_t account,
+                              const mojom::CardanoKeyId& key_id,
+                              base::span<const uint8_t> message) {
+  auto hd_key = DeriveKey(account, key_id);
+  if (!hd_key) {
+    return std::nullopt;
+  }
+  auto signature = hd_key->Sign(message);
+  if (!signature) {
+    return std::nullopt;
+  }
+
+  std::array<uint8_t, kCardanoSignatureSize> result;
+  base::span(result).first<kEd25519PublicKeySize>().copy_from(
+      hd_key->GetPublicKeyAsSpan());
+  base::span(result).last<kEd25519SignatureSize>().copy_from(*signature);
```

Comment:
We can use `span`'s `copy_from_nonoverlapping` for these copy operations.

Reply:
Fixed.